addons/sale_timesheet/models/account.py

```diff
@@ -57,21 +57,15 @@ def _check_sale_line_in_project_map(self):
     @api.multi
     def write(self, values):
         # prevent to update invoiced timesheets if one line is of type delivery
-        if self.sudo().filtered(lambda aal: aal.so_line.product_id.invoice_policy == "delivery") and self.filtered(lambda timesheet: timesheet.timesheet_invoice_id):
-            if any([field_name in values for field_name in ['unit_amount', 'employee_id', 'project_id', 'task_id', 'so_line', 'amount', 'date']]):
-                raise UserError(_('You can not modify already invoiced timesheets (linked to a Sales order items invoiced on Time and material).'))
+        self._check_can_write(values)
         result = super(AccountAnalyticLine, self).write(values)
         return result
 
-    @api.model
-    def _timesheet_preprocess(self, values):
-        values = super(AccountAnalyticLine, self)._timesheet_preprocess(values)
-        # task implies so line (at create)
-        if 'task_id' in values and not values.get('so_line') and values.get('employee_id'):
-            task = self.env['project.task'].sudo().browse(values['task_id'])
-            employee = self.env['hr.employee'].sudo().browse(values['employee_id'])
-            values['so_line'] = self._timesheet_determine_sale_line(task, employee).id
-        return values
+    @api.multi
+    def _check_can_write(self, values):
+        if self.sudo().filtered(lambda aal: aal.so_line.product_id.invoice_policy == "delivery") and self.filtered(lambda timesheet: timesheet.timesheet_invoice_id):
+            if any([field_name in values for field_name in ['unit_amount', 'employee_id', 'project_id', 'task_id', 'so_line', 'amount', 'date']]):
+                raise UserError(_('You can not modify already invoiced timesheets (linked to a Sales order items invoiced on Time and material).'))
```

Comment:
item instead of items?
Have you checked if the term is also in the `.pot` file by the way?

Reply:
You can see on 33092, where the diff is right, that the error message has not been modified.
This PR will be closed automatically when the other will be merged.

Reply:
Ahh yes indeed. Thanks!